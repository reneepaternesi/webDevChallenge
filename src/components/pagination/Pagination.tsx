// React
import React from 'react'
import { PaginationWrapper, PaginationBtn, PaginationLabel } from './styles'

const ITEMS_PER_PAGE = 10

interface Props {
  count?: number
  next?: string
  previous?: string
  currentPage?: number
  onPageChanged: (url: string, newPage: number) => void | Promise<void>
}

const Pagination: React.FC<Props> = ({
  count = 0,
  next = '',
  previous = '',
  currentPage = 1,
  onPageChanged = (url: string) => {},
}) => {
  const totalItems = Math.ceil(count / ITEMS_PER_PAGE)

  const handleClickPrev = (url: string, evt: any) => {
    evt.preventDefault()
    const newPage = currentPage > 0 ? currentPage - 1 : currentPage
    onPageChanged(url, newPage)
  }

  const handleClickNext = (url: string, evt: any) => {
    evt.preventDefault()
    const newPage = currentPage < count ? currentPage + 1 : currentPage
    onPageChanged(url, newPage)
  }

  return (
    <PaginationWrapper>
      <PaginationBtn className= "prev-btn" onClick={(e) => handleClickPrev(previous, e)} disabled={!previous ? true : false}>
        &laquo;
      </PaginationBtn>
      <PaginationLabel className="label">
        Page {currentPage} of {totalItems}
      </PaginationLabel>
      <PaginationBtn className= "next-btn" onClick={(e) => handleClickNext(next, e)} disabled={!next ? true : false}>
        &raquo;
      </PaginationBtn>
    </PaginationWrapper>
  )
}

export default Pagination
