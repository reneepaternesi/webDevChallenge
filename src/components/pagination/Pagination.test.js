import React from 'react'
import { shallow, mount } from 'enzyme'
import Pagination from './Pagination'

describe('Pagination Test', () => {
  const props = {
    count: 60,
    next: 'http://swapi.dev/api/planets/?page=2',
    previous: null,
    currentPage: 1,
    onPageChanged: jest.fn(),
  }

  let wrapper

  beforeEach(() => {
    wrapper = mount(<Pagination {...props} />)
  })
  it('should render Pagination correctly', () => {
    wrapper = shallow(<Pagination {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should has prev button disabled', () => {
    const prevBtn = wrapper.find('button.prev-btn')
    expect(prevBtn.props().disabled).toBe(true)
  })

  it('should has 6 pages', () => {
    const label = wrapper.find('span.label')
    expect(label.text()).toBe('Page 1 of 6')
  })
})
