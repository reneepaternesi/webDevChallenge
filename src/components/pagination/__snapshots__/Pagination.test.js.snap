// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Test should render Pagination correctly 1`] = `
<styled.div>
  <styled.button
    className="prev-btn"
    disabled={true}
    onClick={[Function]}
  >
    «
  </styled.button>
  <styled.span
    className="label"
  >
    Page 
    1
     of 
    6
  </styled.span>
  <styled.button
    className="next-btn"
    disabled={false}
    onClick={[Function]}
  >
    »
  </styled.button>
</styled.div>
`;
